CorpusCallosum/data/fsaverage_cc_template.py

```diff
@@ -0,0 +1,153 @@
+
+from CorpusCallosum.shape.cc_postprocessing import process_slice
+
+
+def smooth_contour(contour, window_size=5):
+        """
+        Smooth a contour using a moving average filter.
+        
+        Parameters
+        ----------
+        contour : tuple of arrays
+            The contour coordinates (x, y).
+        window_size : int
+            Size of the smoothing window.
+            
+        Returns
+        -------
+        tuple of arrays
+            The smoothed contour coordinates (x, y).
+        """
+        x, y = contour
+        
+        # Ensure the window size is odd
+        if window_size % 2 == 0:
+            window_size += 1
+        
+        # Create a padded version of the arrays to handle the edges
+        x_padded = np.pad(x, (window_size//2, window_size//2), mode='wrap')
+        y_padded = np.pad(y, (window_size//2, window_size//2), mode='wrap')
+        
+        # Apply moving average
+        x_smoothed = np.zeros_like(x)
+        y_smoothed = np.zeros_like(y)
+        
+        for i in range(len(x)):
+            x_smoothed[i] = np.mean(x_padded[i:i+window_size])
+            y_smoothed[i] = np.mean(y_padded[i:i+window_size])
+        
+        return (x_smoothed, y_smoothed)
+
+def load_fsaverage_cc_template():
+    # smooth outside contour
+    # Apply smoothing to the outside contour using a moving average
+
+    try:
+        freesurfer_home = Path(os.environ['FREESURFER_HOME'])
+    except KeyError as err:
+        raise OSError(f"FREESURFER_HOME environment variable is not set correctly or does not exist: "
+                      f"{freesurfer_home}, either provide your own template or set the "
+                      f"FREESURFER_HOME environment variable") from err
+
+    fsaverage_seg_path = freesurfer_home / 'subjects' / 'fsaverage' / 'mri' / 'aparc+aseg.mgz'
+    fsaverage_seg = nib.load(fsaverage_seg_path)
+    segmentation = fsaverage_seg.get_fdata()
+
+    PC = np.array([131, 99])
+    AC = np.array([135, 130])
+
+
+    midslice = segmentation.shape[0]//2 +1
+
+    cc_mask = segmentation[midslice] == 251
+    cc_mask |= segmentation[midslice] == 252
+    cc_mask |= segmentation[midslice] == 253
+    cc_mask |= segmentation[midslice] == 254
+    cc_mask |= segmentation[midslice] == 255
+
+    # Smooth the CC mask to reduce noise and irregularities
+
+    # Apply binary closing to fill small holes
+    cc_mask_smoothed = ndimage.binary_closing(cc_mask, structure=np.ones((3, 3)))
+
+    # Apply binary opening to remove small isolated pixels
+    cc_mask_smoothed = ndimage.binary_opening(cc_mask_smoothed, structure=np.ones((2, 2)))
+
+    # Apply Gaussian smoothing and threshold to get a binary mask again
+    cc_mask_smoothed = ndimage.gaussian_filter(cc_mask_smoothed.astype(float), sigma=0.8)
+    cc_mask_smoothed = cc_mask_smoothed > 0.5
+
+    # Use the smoothed mask for further processing
+    cc_mask = cc_mask_smoothed.astype(int)
+    cc_mask[cc_mask > 0] = 192
```

Comment:
This should be faster, no indexing
```suggestion
    cc_mask = cc_mask_smoothed.astype(int) * 192
```